### DISCUSSION TITLE
Data type / metric type mismatch

### DISCUSSION THREAD

**Discussion by Adam-Krull** [milvus-io/milvus#36651] [Q&A and General discussion]
I am building a vector database using embeddings created by an OpenAI text embeddings model. I am generating floating point vectors of small negative and positive numbers. My schema looks like this:

const schema = [
    {
        name: "primary_key",
        description: "primary key",
        data_type: DataType.Int64,
        is_primary_key: true,
        autoID: true,
    },
    {
        name: "vector",
        description: "text vector",
        data_type: DataType.FloatVector,
        dim: dim,
    },
    {
        name: "raw",
        description: "raw text for vector",
        data_type: DataType.VarChar,
        max_length: 1000,
    }
];

and my collection index configuration looks like this: 

const res = await milvusClient.createIndex({
            collection_name: orgName,
            field_name: "vector",
            index_name: "vector_index",
            index_type: "FLAT",
            metric_type: "L2",
});

When I generate an embedding for my query and attempt to search for the most similar vector in my database like this:

const res = await milvusClient.search({
            collection_name: orgName,
            vector: vectorQuery,
            limit: 1,
            output_fields: ["raw"],
});

I get an error message that says there was a mismatch between my data type and metric type. Exact error message pasted below:

{
  status: {
    extra_info: {},
    error_code: 'UnexpectedError',
    reason: 'fail to search on all shard leaders, err=All attempts results:\n' +
      'attempt #1:code: UnexpectedError, error: fail to Search, QueryNode ID=2, reason=stream operation failed: [UnexpectedError] Assert "is_float_data_type == is_float_metric_type" at /go/src/github.com/milvus-io/milvus/internal/core/src/query/SearchBruteForce.cpp:31\n' +
      ' => [BruteForceSearch] Data type and metric type mis-match\n' +
      'attempt #2:context canceled\n',
    code: 0,
    retriable: false,
    detail: ''
  },
  results: []
}

If anyone can help me overcome this error, I will greatly appreciate it. Additional info that may be useful: I am getting embeddings from the OpenAI text-embedding-3-small model. I am specifically requesting vectors of length 128. This constraint is reiterated in my vector database schema. For the record, I have tried the entire tutorial located on the README for this project's Node SDK and I get the same error message there as well. Been stuck on this all day and would love some help. Thank you!

**Comment by xiaofan-luan** [comment]
@shanghaikid 
can you help on verifying this? from the code it seems to be all good to me

**Comment by shanghaikid** [comment]
what is your milvus version? I don't see the data insert part of the code.  can you show me that part?

**Reply by shanghaikid** [reply]
your data structure should be like this: 
https://github.com/milvus-io/milvus-sdk-node/blob/main/examples/milvus/Data.ts
